ambry-quota/src/main/java/com/github/ambry/quota/SimpleQuotaRecommendationMergePolicy.java

```diff
@@ -53,27 +54,24 @@ public SimpleQuotaRecommendationMergePolicy(QuotaConfig quotaConfig) {
 
   @Override
   public ThrottlingRecommendation mergeEnforcementRecommendations(List<QuotaRecommendation> quotaRecommendations) {
-    boolean shouldThrottle = false;
+    QuotaAction mergedQuotaAction = QuotaAction.ALLOW;
     Map<QuotaName, Float> quotaUsagePercentage = new HashMap<>();
-    HttpResponseStatus recommendedHttpStatus = QuotaRecommendationMergePolicy.ACCEPT_HTTP_STATUS;
     long retryAfterMs = DEFAULT_RETRY_AFTER_MS;
     for (QuotaRecommendation recommendation : quotaRecommendations) {
-      boolean currentQuotaShouldThrottle = recommendation.shouldThrottle();
+      QuotaAction currentQuotaAction = recommendation.getQuotaAction();
       if (recommendation.getQuotaName() == QuotaName.READ_CAPACITY_UNIT
           || recommendation.getQuotaName() == QuotaName.WRITE_CAPACITY_UNIT) {
-        currentQuotaShouldThrottle &= quotaConfig.requestThrottlingEnabled;
+        currentQuotaAction = quotaConfig.requestThrottlingEnabled ? currentQuotaAction : QuotaAction.ALLOW;
       } else if (recommendation.getQuotaName() == QuotaName.STORAGE_IN_GB) {
-        currentQuotaShouldThrottle &= quotaConfig.storageQuotaConfig.shouldThrottle;
+        currentQuotaAction = quotaConfig.storageQuotaConfig.shouldThrottle ? currentQuotaAction : QuotaAction.ALLOW;
```

Comment:
This is sort of because the way we developed. You are right in pointing out that this needs refactoring. I will create a separate PR out for refactoring the quota configs. There are some configs which can be simplified into one, and there are a few that can just be removed.